spec.html

```diff
@@ -7652,19 +7650,44 @@ <h1>Jobs and Host Operations to Enqueue Jobs</h1>
     </emu-note>
 
+    <p>An invocation of a Job <dfn id="job-propagatesactivescriptormodule">propagates the script or module</dfn> _scriptOrModule_ (a Script Record, a Module Record, or *null*) if all of the following conditions are true:</p>
+    <ul>
+      <li>GetActiveScriptOrModule() is _scriptOrModule_.</li>
+      <li>If _scriptOrModule_ is a Script Record or Module Record, let _ec_ be the topmost execution context on the execution context stack whose ScriptOrModule component is _scriptOrModule_. The Realm component of _ec_ is _scriptOrModule_.[[Realm]].</li>
+    </ul>
+
+    <p>An invocation of a Job is <dfn id="job-preparedtoevaluatecode">prepared to evaluate code</dfn> if all of the following conditions are true:</p>
+    <ul>
+      <li>The execution context stack is not empty.</li>
+      <li>The Realm component of the topmost execution context on the execution context stack is a Realm Record.</li>
+    </ul>
+
+    <emu-note>
+      <p>Host environments may prepare an execution to evaluate code by pushing execution contexts onto the execution context stack. The specific steps are implementation-defined.</p>
+      <p>The specific choice of Realm is up to the host environment. This initial execution context and Realm is only in use before any callback function is invoked. When a callback function related to a Job, like a Promise handler, is invoked, the invocation pushes its own execution context and Realm.</p>
+    </emu-note>
+
+    <p>Particular kinds of Jobs have additional conformance requirements.</p>
+
     <emu-clause id="sec-hostenqueuepromisejob" aoid="HostEnqueuePromiseJob">
       <h1>HostEnqueuePromiseJob ( _job_, _realm_ )</h1>
       <p>HostEnqueuePromiseJob is a host-defined abstract operation that schedules the Job Abstract Closure _job_ to be performed, at some future time. The Abstract Closures used with this algorithm are intended to be related to the handling of Promises, or otherwise, to be scheduled with equal priority to Promise handling operations.</p>
-      <p>The _realm_ parameter is passed through to hosts with no normative requirements; it is either *null* or a Realm.</p>
+      <p>The _realm_ parameter is either *null* or a Realm Record.</p>
+
+      <p>The implementation of HostEnqueuePromiseJob must conform to the requirements in <emu-xref href="#sec-jobs"></emu-xref>. Additionally, Jobs enqueued by this hook must conform to all of the following requirements:</p>
+      <ul>
+        <li>If _realm_ is not *null*, the implementation must perform implementation-defined steps such that all invocations of _job_ are prepared to evaluate code.</li>
+        <li>Let _scriptOrModule_ be GetActiveScriptOrModule(). The implementation must perform implementation-defined steps such that all invocations of _job_ propagate the script or module _scriptOrModule_.</li>
```

Comment:
As above, it would be good to be precise about the time at which `GetActiveScriptOrModule` is being invoked. Here I think it's a little more clearly the point at which `HostEnqueuePromiseJob` is invoked, but I think it would still be good to spell it out.

Reply:
Will clarify.